packages/common-tests/browser-tests/page-model/eventSearchPage.ts

```diff
@@ -0,0 +1,84 @@
+import { screen } from '@testing-library/testcafe';
+import { t } from 'testcafe';
+import { initTestI18n as i18n } from '../../../common-i18n/src';
+import type { AppNamespace } from '../types/app-namespace';
+
+class EventSearchPage {
+  searchText = 'helsinki';
+  private appNamespace: 'appHobbies' | 'appEvents' | 'appSports';
+
+  constructor(appNamespace: AppNamespace) {
+    this.appNamespace = appNamespace;
+  }
+
+  protected get searchFormHeading() {
+    const searchHeadingText = i18n.t(`search:search.labelSearchField`);
+    return screen.getByRole('heading', {
+      name: searchHeadingText,
+    });
+  }
+
+  protected get searchButton() {
+    const searchButtonText = i18n.t(`search:search.buttonSearch`);
+    return screen.getByRole('button', {
+      name: searchButtonText,
+    });
+  }
+
+  protected get autoSuggestInput() {
+    const searchPlaceholderText = i18n.t(
+      `${this.appNamespace}:search.search.placeholder`
+    );
+    return screen.getByPlaceholderText(searchPlaceholderText);
+  }
+
+  protected get buttonLoadMore() {
+    const loadMoreText = i18n.t('search:buttonLoadMore');
+    return screen.queryByRole('button', {
+      name: loadMoreText,
+    });
+  }
+
+  protected get results() {
+    const cardLinkRoleName = i18n
+      .t('event:eventCard.ariaLabelLink')
+      .replace('{{name}}', '*.');
+    return screen.queryAllByRole('link', {
+      name: new RegExp(cardLinkRoleName, 'i'),
+    });
+  }
+
+  protected get notFoundResult() {
+    const notFoundText = i18n.t('search:searchNotification');
+    return screen.queryByText(notFoundText);
+  }
+
+  public async makeSuccessfulSearch() {
```

Comment:
Usually actions and expectations are separated to different methods so I would probably have:
```
public async doSuccessfulSearch() {
...
}
public async expectSearchResults() {
...
}

public async doUnsuccessfulSearch() {
...
}
public async expectNoSearchResults() {
...
}
```

Reply:
Also I probably would call those methods straight from the `search-page.testcafe.ts` test so It's clear what is tested. the `verify()` doesnt tell much and reader probably have to jump into the method to see what it does. Of course this more like opinion, so feel free to ignore :)